bindings/go/src/fdb/snapshot.go

```diff
@@ -86,3 +92,9 @@ func (s Snapshot) GetReadVersion() FutureInt64 {
 func (s Snapshot) GetDatabase() Database {
 	return s.transaction.db
 }
+
+// Snapshot returns the receiver and allows Snapshot to satisfy the
+// ReadTransaction interface.
+func (s Snapshot) Options() TransactionOptions {
+	return s.Options()
+}
```

Comment:
This was copied from the latest FDB go bindings, but this looks like an infinite recursion. Fixing.
Tests to confirm this can be found here.